src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -301,6 +317,83 @@ undistribute_table(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * alter_distributed_table gets a distributed table and some other
+ * parameters and alters some properties of the table according to
+ * the parameters.
+ */
+Datum
+alter_distributed_table(PG_FUNCTION_ARGS)
+{
+	Oid relationId = InvalidOid;
+	relationId = PG_GETARG_OID(0);
+	
+	char *distributionColumn = NULL;
+	if (!PG_ARGISNULL(1))
+	{
+		text *distributionColumnText = PG_GETARG_TEXT_P(1);
+		distributionColumn = text_to_cstring(distributionColumnText);
+	}
+
+	int shardCount = 0;
+	bool shardCountIsNull = true;
+	if (!PG_ARGISNULL(2))
+	{
+		shardCount = PG_GETARG_INT32(2);
+		shardCountIsNull = false;
+	}
+
+	char *colocateWith = NULL;
+	if (!PG_ARGISNULL(3))
+	{
+		text *colocateWithText = PG_GETARG_TEXT_P(3);
+		colocateWith = text_to_cstring(colocateWithText);
+	}
+
+	bool cascadeToColocated = false;
+	bool cascadeToColocatedIsNull = true;
+	if (!PG_ARGISNULL(4))
+	{
+		cascadeToColocated = PG_GETARG_BOOL(4);
+		cascadeToColocatedIsNull = false;
+	}
+
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
+	EnsureRelationExists(relationId);
+	EnsureTableOwner(relationId);
+
+	AlterDistributedTableMessages(relationId, distributionColumn, shardCountIsNull, shardCount, colocateWith, cascadeToColocatedIsNull, cascadeToColocated);
+
+	AlterDistributedTable(relationId, distributionColumn, shardCount, colocateWith, cascadeToColocated);
+
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * alter_table_set_access_method gets a distributed table and an access
+ * method and changes table's access method into that.
+ */
+Datum
+alter_table_set_access_method(PG_FUNCTION_ARGS)
+{
+	Oid relationId = PG_GETARG_OID(0);
+	
+	text *accessMethodText = PG_GETARG_TEXT_P(1);
+	char *accessMethod = text_to_cstring(accessMethodText);
+
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
```

Comment:
minor: it would be nicer to move this check to CreateDistributedTable to allow this function to be used on shards and error out only if we're wroking on a Citus table